lib/cache/collections.go

```diff
@@ -1346,38 +1347,27 @@ func (s *databaseService) erase(ctx context.Context) error {
 }
 
 func (s *databaseService) fetch(ctx context.Context) (apply func(ctx context.Context) error, err error) {
+	resources, err := client.GetResourcesWithFilters(ctx, s.Presence, proto.ListResourcesRequest{ResourceType: types.KindDatabaseService})
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
```

Comment:
```suggestion
	databaseServices, err := types.ResourcesWithLabels(resources).AsDatabaseServices()
	if err != nil {
		return nil, trace.Wrap(err)
	}

```

And then you can iterate over `databaseServices`.

Reply:
Yeah, we're not really expecting that to fail anyway.